File: xLiAd.DiagnosticLogCenter.Abstract/TracePageIdValue.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace xLiAd.DiagnosticLogCenter.Abstract
{
    public class TracePageIdValue
    {
        public TracePageIdValue(DateTime happenTime, string clientName, string envName, Guid guid)
        {
            HappenTime = happenTime;
            ClientName = clientName;
            EnvName = envName;
            Guid = guid;
        }
        public TracePageIdValue(DateTime happenTime, string clientName, string envName) : this(happenTime, clientName, envName, Guid.NewGuid()) { }

        public DateTime HappenTime { get; }
        public string ClientName { get; }
        public string EnvName { get; }
        public Guid Guid { get; }

        public override string ToString()
        {
            return HappenTime.ToString("yyyyMMdd-HHmmss-fff-") + ClientName + "-" + EnvName + "-" + Guid.ToString();
        }

        public static TracePageIdValue FromString(string s)
        {
            if (string.IsNullOrEmpty(s))
                throw new ArgumentNullException("s", "试图转化 null/空串 到一个 TraceId/PageId 。");
            var sa = s.Split('-');
            //if (sa.Length != 6)
            //    throw new ArgumentException($"转化一个 TraceId/PageId 时提供了一个不合规的字符串，应为6段，实为：{sa.Length}段。", "s");
            if(sa[0].Length != 8 || sa[1].Length != 6 || !int.TryParse(sa[0], out var ymd) || !int.TryParse(sa[1], out var hms) || !int.TryParse(sa[2], out var fff))
                throw new ArgumentException($"转化一个 TraceId/PageId 时提供了一个不合规的字符串{s}。", "s");
            DateTime dt = new DateTime(ymd / 10000, (ymd % 10000) / 100, ymd % 100, hms / 10000, (hms % 10000) / 100, hms % 100, fff);
            var result = new TracePageIdValue(dt, sa[3], sa[4], Guid.Parse(string.Join("-",sa.Skip(5))));
            return result;
        }
    }
}

File: DiagnosticLogCenter-Java/src/main/java/com/xliad/diagnosticlogcenter/abstract/TracePageIdValue.java
package com.xliad.diagnosticlogcenter.abstracts;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.Arrays;
import java.util.UUID;

public class TracePageIdValue {
    private final LocalDateTime happenTime;
    private final String clientName;
    private final String envName;
    private final UUID guid;

    public TracePageIdValue(LocalDateTime happenTime, String clientName, String envName, UUID guid) {
        this.happenTime = happenTime;
        this.clientName = clientName;
        this.envName = envName;
        this.guid = guid;
    }

    public TracePageIdValue(LocalDateTime happenTime, String clientName, String envName) {
        this(happenTime, clientName, envName, UUID.randomUUID());
    }

    public LocalDateTime getHappenTime() {
        return happenTime;
    }

    public String getClientName() {
        return clientName;
    }

    public String getEnvName() {
        return envName;
    }

    public UUID getGuid() {
        return guid;
    }

    @Override
    public String toString() {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMMdd-HHmmss-SSS");
        return happenTime.format(formatter) + "-" + clientName + "-" + envName + "-" + guid.toString();
    }

    public static TracePageIdValue fromString(String s) {
        if (s == null || s.isEmpty()) {
            throw new IllegalArgumentException("试图转化 null/空串 到一个 TraceId/PageId");
        }
        String[] sa = s.split("-");
        if (sa.length < 6) {
            throw new IllegalArgumentException("转化一个 TraceId/PageId 时提供了一个不合规的字符串: " + s);
        }

        // 解析日期时间
        String datePart = sa[0] + sa[1] + sa[2];
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMMddHHmmssSSS");
        LocalDateTime dt = LocalDateTime.parse(datePart, formatter);

        // 组合剩余的GUID部分
        String guidStr = String.join("-", Arrays.copyOfRange(sa, 5, sa.length));

        return new TracePageIdValue(dt, sa[3], sa[4], UUID.fromString(guidStr));
    }
}